store/mongo/migration_0_3_0.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mongo
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	mstore "github.com/mendersoftware/go-lib-micro/store"
+	"github.com/mendersoftware/inventory/model"
+)
+
+type migration_0_3_0 struct {
+	ms  *DataStoreMongo
+	ctx context.Context
+}
+
+// Up creates timestamp and group attributes under the identity scope.
+// The values reflect the values previously held in the root of the document.
+func (m *migration_0_3_0) Up(from migrate.Version) error {
+	const identityScope = DbDevAttributes + "." + model.AttrScopeIdentity
+
+	databaseName := mstore.DbFromContext(m.ctx, DbName)
+	collDevs := m.ms.client.Database(databaseName).Collection(DbDevicesColl)
+	upsertOpts := mopts.Update().SetUpsert(true)
+
+	// Add timestamp attributes to the identity scope
+	cursor, err := collDevs.Find(m.ctx, bson.M{})
+	if err != nil {
+		return err
+	}
+	defer cursor.Close(m.ctx)
+	for cursor.Next(m.ctx) {
```

Comment:
ok, I am against running update for each device. this can be done with single UpdateMany query with rename and set. also: I do not think that leaving group field is a good idea, it cant be changed or accessed by means of API calls.

Reply:
Yeah, I think there's room for improvement here. I'm not sure if what you're suggesting is possible though, I don't believe it's possible for an update $set operation to use other fields of the document as value (at least not with version 3.6).